src/App.js

```diff
@@ -3,26 +3,29 @@ import React, { useEffect } from 'react';
 import './App.css';
 import Header from './Header';
 import { Feed } from './features/feed/Feed';
-import HotTopic from './HotTopic';
-import { useDispatch } from 'react-redux';
+import HotTopic from './features/HotTopic/HotTopic';
+import { useDispatch, useSelector } from 'react-redux';
 import { setPosts } from './features/feed/feedSlice';
 
 function App() {
+
+    const selectSub = useSelector(state => state.sub)
+
     const dispatch = useDispatch();
     const limit = 5;
     const baseUrl = 'https://www.reddit.com';
 
     useEffect(() => {
         const getApiData = async () => {
             const response = await fetch(
-                `${baseUrl}/r/popular.json?limit=${limit}`
+                `${baseUrl}/r/${selectSub.sub}.json?limit=${limit}`
             ).then((response) => response.json());
 
             dispatch(setPosts(response.data.children));
         };
 
         getApiData();
-    }, []);
+    }, [selectSub.sub]);
```

Comment:
Adding dispatch to the dependency should be fine.
https://stackoverflow.com/a/66568609